### PR TITLE
py-poxy: add new package

### PR DESCRIPTION
**Pull request by pranav-sivaraman**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-poxy/package.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    license("MIT", checked_by="pranav-sivaraman")
+
+    version("0.18.0", sha256="f5da8ff04ec08859bfd1c8ec6ef61b70e3af630915a4cce6a3e377eec3bcd3d4")
+
+    depends_on("py-setuptools", type="build")
+
+    with default_args(type=("build", "run")):
+        depends_on("python@3.7:")
+        depends_on("py-misk")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-poxy/package.py]: Should this be `depends_on("py-misk@0.8.1:")`?

From https://github.com/marzer/poxy/blob/main/pyproject.toml:
```
...
dependencies = [
    'misk >= 0.8.1',
    'bs4',
    'jinja2',
    'pygments',
    'html5lib',
    'lxml',
    'tomli',
    'schema != 0.7.5',
    'requests',
    'trieregex',
    'colorama',
]
...
```

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage and version sha256 but cannot `spack stage` or `spack fetch` the package (see below).  Are you able to successfully build the package using Spack?

```
==> Error: Couldn't find patch for package builtin.libgcrypt with sha256: 4e308ba07a5065c052aba99b8c0e85a77b9bdccc0ea6aa975dde22f076a3fb73. This usually means the patch was modified or removed. To fix this, either reconcretize or use the original package repository
```

File: var/spack/repos/builtin/packages/py-poxy/package.py (hunk 1 above)
Comment (pranav-sivaraman) [reply var/spack/repos/builtin/packages/py-poxy/package.py]: Yes. Technically `py-misk@0.8.1` is the only version that exists in the builtin repo, but I'll make the change.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-poxy/package.py]: People have come along later and added older versions to packages so being explicit is generally safer.

**Comment by pranav-sivaraman** [comment]
> Are you able to successfully build the package using Spack?

I was able to build this package yesterday on macOS Sonoma using apple-clang. Let me double check.

**Comment by pranav-sivaraman** [comment]
I just did a full rebuild of the entire spec for `py-poxy` on my machine and it works just fine (`import poxy` works). Not sure what's causing the staging issues on your end.

**Comment by pranav-sivaraman** [comment]
I think you need to approve the PR for the auto-merge to kick in?

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**